### PR TITLE
feat: create model transformers for VC and VP

### LINKED ISSUE #3501: IATP: create model classes and transformers
# Feature Request

This issue is intended as preparatory work for the IATP implementations, it calls for the implementation of the following model classes:
- `VerifiableCredential`
- `VerifiablePresentation`

these go into an SPI module together with the interfaces.

Other objects we need beforehand include JSON-LD transformers, which should go in a separate module.

## Which Areas Would Be Affected?

iam

## Why Is the Feature Desired?

IATP adoption

## Solution Proposal

_If possible, provide a (brief!) solution proposal._

### PR DESCRIPTION
**Pull request by paullatzelsperger**
## What this PR changes/adds

This PR adds transformers that can convert JSON-LD -> `VerifiableCredential` and `VerifiablePresentation`.

## Why it does that

Preliminary work needed for subsequent IATP adoption tasks.

## Further notes

- The `VerifiableCredential` itself does **not** hold the raw VC anymore
- Introduced a `VerifiableCredentialContainer` (name is up for discussion), that stores the raw VC (JWT or JSON-LD), the credential format and the deserialized `VerifiableCredential` instance.

## Linked Issue(s)

Closes #3501

_Please be sure to take a look at the [contributing guidelines](https://github.com/eclipse-edc/.github/blob/main/CONTRIBUTING.md#submit-a-pull-request) and our [etiquette for pull requests](https://github.com/eclipse-edc/.github/blob/main/contributing/pr_etiquette.md)._

### CODE REVIEW

File: extensions/common/iam/identity-trust/identity-trust-transform/src/main/java/org/eclipse/edc/iam/identitytrust/IdentityTrustTransformExtension.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+    @Inject
+    private TypeTransformerRegistry typeTransformerRegistry;
+
+    @Inject
+    private JsonLd jsonLdService;
+
+    @Override
+    public void initialize(ServiceExtensionContext context) {
+        ServiceExtension.super.initialize(context);
```
Comment (bscholtes1A) [inline extensions/common/iam/identity-trust/identity-trust-transform/src/main/java/org/eclipse/edc/iam/identitytrust/IdentityTrustTransformExtension.java:45]: ?

File: spi/common/identity-trust-spi/src/main/java/org/eclipse/edc/identitytrust/model/VerifiablePresentation.java [hunk 2]
```diff
@@ -17,23 +17,21 @@
...
 
+import static org.eclipse.edc.identitytrust.VcConstants.VC_PREFIX;
+
 /**
  * Represents a VerifiablePresentation object as specified by the
  * <a href="https://w3c.github.io/vc-data-model/#presentations-0">W3 specification</a>
  */
 public class VerifiablePresentation {
-    public static final String DEFAULT_TYPE = "VerifiablePresentation";
-    private final String rawVp;
```
Comment (bscholtes1A) [inline spi/common/identity-trust-spi/src/main/java/org/eclipse/edc/identitytrust/model/VerifiablePresentation.java:26]: Will there be a `VerifiablePresentationContainer` as well?

File: extensions/common/iam/identity-trust/identity-trust-transform/src/main/java/org/eclipse/edc/iam/identitytrust/IdentityTrustTransformExtension.java (hunk 1 above)
Comment (paullatzelsperger) [reply extensions/common/iam/identity-trust/identity-trust-transform/src/main/java/org/eclipse/edc/iam/identitytrust/IdentityTrustTransformExtension.java:45]: remnant of IntelliJ's code generation...

File: spi/common/identity-trust-spi/src/main/java/org/eclipse/edc/identitytrust/model/VerifiablePresentation.java (hunk 2 above)
Comment (paullatzelsperger) [reply spi/common/identity-trust-spi/src/main/java/org/eclipse/edc/identitytrust/model/VerifiablePresentation.java:26]: yep, it exists - on my machine. Simply hadn't staged it for commit.

**Review by bscholtes1A — APPROVED** [review]

**Review by wolf4ood — APPROVED** [review]

**Pull request merged**